### PR TITLE
385/lintissues

### PR DESCRIPTION
**Pull request by Haashi**
#385 
New staticcheck reports : 
```
d2common/d2data/d2datadict/experience.go:40:5: var experienceStringMap is unused (U1000)
d2common/d2data/d2datadict/experience.go:41:5: var experienceHeroMap is unused (U1000)
d2core/d2gui/manager.go:155:19: func (*manager).clear is unused (U1000)
d2core/d2map/d2mapentity/animated_entity.go:12:2: field action is unused (U1000)
d2core/d2map/d2mapentity/animated_entity.go:13:2: field repetitions is unused (U1000)
d2core/d2map/d2maprenderer/viewport.go:26:2: field align is unused (U1000)
d2core/d2map/d2maprenderer/viewport.go:149:20: func (*Viewport).toLeft is unused (U1000)
d2core/d2map/d2maprenderer/viewport.go:157:20: func (*Viewport).toRight is unused (U1000)
d2core/d2map/d2maprenderer/viewport.go:166:20: func (*Viewport).resetAlign is unused (U1000)
d2core/d2render/ebiten/composite_mode_helper.go:41:6: func ebitenToD2CompositeMode is unused (U1000)
d2core/d2render/ebiten/filter_helper.go:21:6: func ebitenToD2Filter is unused (U1000)
d2game/d2gamescreen/map_engine_testing.go:91:2: field debugVisLevel is unused (U1000)
d2game/d2player/game_controls.go:237:2: this value of w is never used (SA4006)
d2game/d2player/game_controls.go:264:2: this value of w is never used (SA4006)
d2game/d2player/inventory.go:119:2: this value of w is never used (SA4006)
d2game/d2player/inventory.go:126:2: this value of w is never used (SA4006)
d2game/d2player/inventory_grid.go:111:3: this value of err is never used (SA4006)
d2game/d2player/stats.go:68:2: this value of w is never used (SA4006)
d2game/d2player/stats.go:75:2: this value of w is never used (SA4006)
d2game/d2player/stats.go:89:2: this value of w is never used (SA4006)
d2game/d2player/stats.go:107:2: this value of w is never used (SA4006)
d2game/d2player/stats.go:114:2: this value of w is never used (SA4006)
d2networking/d2client/d2remoteclient/remote_client_connection.go:115:3: this value of err is never used (SA4006)
d2networking/d2client/d2remoteclient/remote_client_connection.go:117:3: this value of err is never used (SA4006)
d2networking/d2server/game_server.go:91:3: this value of err is never used (SA4006)
d2networking/d2server/game_server.go:93:3: this value of err is never used (SA4006)
```

Most of the fixes are quite small. The only important parts are the rewrite of item_affix.go and the label jump on huffman.go (couldn't think of a better solution).

### CODE REVIEW

File: d2common/d2data/d2compression/huffman.go [hunk 1]
```diff
@@ -347,23 +347,19 @@ func HuffmanDecompress(data []byte) []byte {
...
 	var decoded int
-	for true {
+Loop:
+	for {
 		node := decode(bitstream, head)
 		decoded = node.DecompressedValue
 		switch decoded {
 		case 256:
-			break
+			break Loop
```
Comment (mewmew) [inline d2common/d2data/d2compression/huffman.go:356]: nice catch

File: d2common/d2data/d2datadict/item_affix.go [hunk 2]
```diff
@@ -109,11 +107,11 @@ func loadDictionary(
 
 func createItemAffixRecords(
 	d *d2common.DataDictionary,
-	r []*ItemAffixCommonRecord,
```
Comment (mewmew) [inline d2common/d2data/d2datadict/item_affix.go:112]: I haven't looked at the details for where memory is allocated in OpenDiablo2, but sometimes passing in an existing slice may help us reuse the underlying backing array, so we don't have to allocate memory multiple times if we invoke `createItemAffixRecords` more than once.

File: d2core/d2render/ebiten/ebiten_renderer.go [hunk 3]
```diff
@@ -31,9 +31,9 @@ func CreateRenderer() (*Renderer, error) {
 
 	config := d2config.Get()
 
-	ebiten.SetCursorVisible(false)
+	ebiten.SetCursorMode(ebiten.CursorModeHidden)
```
Comment (mewmew) [inline d2core/d2render/ebiten/ebiten_renderer.go:34]: Good that you replaced depricated functions!

Comment (essial) [reply d2core/d2render/ebiten/ebiten_renderer.go:34]: I wonder if that fixes the issue of the cursor not hiding when toggling fullscreen on OSX..

Comment (mewmew) [reply d2core/d2render/ebiten/ebiten_renderer.go:34]: > I wonder if that fixes the issue of the cursor not hiding when toggling fullscreen on OSX..

Not likely. The deprecated functions of ebiten still invoke the new functions underneath.

```
$ go doc -src ebiten.SetCursorVisible

package ebiten // import "github.com/hajimehoshi/ebiten"

// SetCursorVisible is deprecated as of 1.11.0-alpha. Use SetCursorMode instead.
func SetCursorVisible(visible bool) {
	if visible {
		SetCursorMode(CursorModeVisible)
	} else {
		SetCursorMode(CursorModeHidden)
	}
}
```

File: d2common/d2data/d2datadict/item_affix.go (hunk 2 above)
Comment (Haashi) [reply d2common/d2data/d2datadict/item_affix.go:112]: Good point. But this fonction is actually called twice (once for magic prefix and once for magic suffix) with around 750 items when the game launches. I don't think the underlying array is ever reused anyway.

**Comment by essial** [comment]
Looks good to me, anyone else have any comments before I merge?

**Pull request merged**